### PR TITLE
feat: allow challenges to be hidden by language

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
@raisedadead this fails if there are missing challenges for a language (like espanol currently), but once they're synced from Crowdin it should be fine.

Since no certs are 100% proofread this PR currently defaults to English no matter what.  Also, we don't have a Crowdin project for certifications (the .markdown files) yet, so they also default to English.

### CODE REVIEW

**Review by naomi-lgbt — DISMISSED** [review]
Looks good to me.

Tested this locally, adding `responsive-web-design` to the Chinese array - it successfully rendered just that superblock with the Chinese files and the rest appeared as English.

File: utils/is-audited.js
Comment (raisedadead) [inline utils/is-audited.js:21]: Leys drop these comments - classic example of how comments go stale. I much rather not comment vs letting them go stale.

**Comment by raisedadead** [comment]
Hey @RandellDawson could you confirm that we would want a dedicated project for certs on Crowdin or it would be possible to club them together with the curriculum.

The goal here is the least involvement in maintenance efforts from the Staff's perspective.

File: utils/is-audited.js
Comment (ojeytonwilliams) [reply utils/is-audited.js:21]: I get your point, but code can go stale, too.

I'll do some reading around the topic, but I feel like the cost of having to catch this in QA (like this) is outweighed by the benefit that the comment provides.

Comment (raisedadead) [reply utils/is-audited.js:21]: Lines 20-21 as [seen here](https://github.com/freeCodeCamp/freeCodeCamp/commit/6269f783cc7291b75af3aac6e62890f1bac81e33#diff-3a4b75b56b44645bb3c05d7031cf913278c9efd2fe4d78a1f6518555ac3dbe26R20-R21) are now stale. I am not talking about lines 3-10 as seen here [https://github.com/freeCodeCamp/freeCodeCamp/commit/6269f783cc7291b75af3aac6e62890f1bac81e33#diff-3a4b75b56b44645bb3c05d7031cf913278c9efd2fe4d78a1f6518555ac3dbe26R3-R10]

Comment (ojeytonwilliams) [reply utils/is-audited.js:21]: Yeah, I got that. I agree that these stale ones should go. It's possible were talking about different things - I thought you were arguing that comments than *can* go stale are bad.

Comment (raisedadead) [reply utils/is-audited.js:21]: Let's agree to disagree. 

The point being, at the end of the day as a maintainer - it's us who are put at blame by the regardless of how much efforts you put in to make lives of the contributors easier. 

I do not care for comments in code, others do - and that's okay. There are well established patterns in out there to maintain code properly, I personally do not have had the time because this is bike shedding territory. Any documentation that we do have has been very painstakingly done (with significant contributions from me too. So, I get why we should have code comments).

The only thing that I do care about is the unfathomable tech-debt, and the code quality. Those are my highest priorities over code comments and documentation.

**Comment by RandellDawson** [comment]
> Hey @RandellDawson could you confirm that we would want a dedicated project for certs on Crowdin or it would be possible to club them together with the curriculum.
> 
> The goal here is the least involvement in maintenance efforts from the Staff's perspective.

I can include the certificates folder in Crowdin, but the `yaml` section gets treated as a string, so we will not be able to hide anything inside the `yaml` section.  Ideally, we would be able to hide everything that is not the front matter `title` or the `title properties of each `test`.  If that will suffice for now, I can modify the existing GitHub action that hides the strings to hide all strings except the front matter `title` key and the `yaml` portion.

@ojeytonwilliams If the certs could be converted to YAML or JSON files, it would be easier to hide what we want on Crowdin.

**Comment by ojeytonwilliams** [comment]
@RandellDawson I'd like to convert to YAML and rip out the old parser entirely, so could you hold off creating the project until that's done?  We should also try uploading a few test .yaml certs to make sure there are no surprises waiting for us.

**Review by raisedadead — APPROVED** [review]

File: utils/is-audited.js
Comment (ojeytonwilliams) [reply utils/is-audited.js]: Understood.  I think this is a conversation worth having, but not now and not buried away here.  Let's pick this up again when we both have a bit more time.

**Pull request merged**
